compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -243,6 +243,17 @@ trait QuotesAndSplices {
         res
       }
 
+      def checkAlternativeBinds(pat0: Tree): Unit =
+        def rec(pat: Tree): Unit =
+          pat match
+            case Typed(pat, _) => rec(pat)
+            case UnApply(_, _, pats) => pats.foreach(rec)
+            case pat: Bind =>
+              report.error(IllegalVariableInPatternAlternative(pat.symbol.name), pat.withSpan(pat.nameSpan))
```

Comment:
How come we need to check this here, isn't typedBind going to run on this tree later and produce the same error via https://github.com/lampepfl/dotty/blob/400427db21410b366e4d9b890d73cec15416b632/compiler/src/dotty/tools/dotc/typer/Typer.scala#L2141-L2142 ?

Reply:
The context at that point lost the mode `InPatternAlternative`. Now I found out where that is happening and will push a better fix.